tests/functional/test_sts.py

```diff
@@ -288,3 +288,27 @@ def test_client_for_unknown_region_with_regional_configured(self):
             expected_url='https://sts.not-real.amazonaws.com/',
             expected_signing_region='not-real',
         )
+
+
+def test_assume_role_with_saml_no_region_custom_endpoint(patched_session):
+    # When an endpoint_url and no region are given, AssumeRoleWithSAML should
+    # resolve to the endpoint_url and succeed, not fail in endpoint resolution:
+    # https://github.com/aws/aws-cli/issues/7455
+
+    client = patched_session.create_client(
+        'sts', region_name=None, endpoint_url="https://custom.endpoint.aws"
+    )
+    mock_response_body = b"""\
+<AssumeRoleWithSAMLResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
+    <AssumeRoleWithSAMLResult></AssumeRoleWithSAMLResult>
+</AssumeRoleWithSAMLResponse>
+"""
```

Comment:
We should also probably add an assertion that the `client.meta.region_name` is `None`. Even though we specify `region_name=None`, we could potential inherit region from the session. Then given the `patched_session` seems like it will be widely used, we could potentially drift from this initial state if for example we decided to add a region to the `patched_session` fixture later on.

Reply:
Done in bbd870b